python/lsst/ctrl/pool/parallel.py

```diff
@@ -404,11 +410,28 @@ def parseAndSubmit(cls, args=None, **kwargs):
         if not cls.RunnerClass(cls, batchArgs.parent).precall(batchArgs.parent):  # Write config, schema
             taskParser.error("Error in task preparation")
 
-        numCores = batchArgs.cores if batchArgs.cores > 0 else batchArgs.nodes*batchArgs.procs
-        walltime = cls.batchWallTime(batchArgs.time, batchArgs.parent, numCores)
+        if batchArgs.batch is None:     # don't use a batch system
+            #
+            # Remove --batch-type argument
+            #
+            i, argv = 0, []
+            while i < len(sys.argv):
+                a = sys.argv[i]
+                if a.startswith("--batch-t"): # i.e. --batch-type and abbreviations thereof
+                    i += 1                    # skip the value
```

Comment:
What if I specify `--batch-type=none` (as `argparse` supports)?  Then the next, unrelated, argument gets dropped.

But you shouldn't be parsing the command-line yourself.  I suggest you use:

    sys.argv = shCommandFromArgs(batchArgs.leftover)

That will strip out all batch-related arguments.